plugin/api.go

```diff
@@ -193,6 +193,9 @@ type API interface {
 	// KVDelete will remove a key-value pair. Returns nil for non-existent keys.
 	KVDelete(key string) *model.AppError
 
+	// KVList will list all keys for a plugin. Returns nil for non-existent keys.
+	KVList() ([]string, *model.AppError)
```

Comment:
@jwilander, thinking about performance here, but should we build in paging from the beginning, or do we anticipate this store to be used only for "small" amounts of data, and perhaps we'd introduce something better in the future?

Reply:
Good point, I don't think we should assume how the plugin is going use the KV store so it makes sense to think about scale from the start and add the paging. Thoughts @gupsho?

Reply:
Makes sense. Will add pagination support.